File: sort/冒泡排序/bubble.lua
-- input:
    -- 4
    -- 1
    -- 5
    -- 2
    -- 3
-- output:
    -- 1
    -- 2
    -- 3
    -- 4
    -- 5

function bubble(array)
	local tpl = 0
	for i=1,#array-1 do
		for j=1,#array-i do
			if array[j] > array[j+1] then
				tpl = array[j]
				array[j] = array[j+1]
				array[j+1] = tpl
			end
		end
	end
end
 
function main( ... )
    local array = {4,1,5,2,3}
    print("input:")
    for i,v in ipairs(array) do
		print(v)
	end
    print("output:")
    bubble(array)
	for i,v in ipairs(array) do
		print(v)
	end
end
 
main()

File: sort/冒泡排序/bubble.cpp
# include <iostream>
using namespace std;

/*
	input:
	4 1 5 2 4
	output:
	1 2 4 4 5
*/
void Bubble(int array[],int len){
	int i,j,tpl;
	for (i = 0;i<len-1;i++)
		for (j = 0; j < len-1; j++)
			if (array[j]>array[j+1]){
				tpl = array[j];
				array[j] = array[j+1];
				array[j+1] = tpl;
			}
}

int main(){
	int array[]={4,1,5,2,3}; 
	int len = sizeof(array)/sizeof(array[0]);
	cout << "input:" << endl;
	for (int i = 0; i < len; i++)
        cout << array[i] << ' ';
	cout << endl;
	// bubble sort 
	Bubble(array,len);
	cout << "output:" << endl;
	for (int i = 0; i < len; i++)
        cout << array[i] << ' ';
	cout << endl;
}

